src/main/java/co/cask/hydrator/salesforce/SObjectsDescribeResult.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.hydrator.salesforce;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Lists;
+import com.sforce.soap.partner.DescribeSObjectResult;
+import com.sforce.soap.partner.Field;
+import com.sforce.soap.partner.PartnerConnection;
+import com.sforce.ws.ConnectionException;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Retrieves {@link DescribeSObjectResult}s for the given sObjects
+ * and adds field information to the internal holder.
+ * This class will be used to populate {@link SObjectDescriptor} for queries by sObject
+ * or to generate CDAP schema based on Salesforce fields information.
+ */
+public class SObjectsDescribeResult {
+
+  // key -> [sObject name], value -> [key -> field name,  value -> field]
+  private final Map<String, Map<String, Field>> holder = new HashMap<>();
+
+  public SObjectsDescribeResult(PartnerConnection connection, Collection<String> sObjects) {
+    // Salesforce limitation that we can describe only 100 sObjects at a time
+    // split the given sObjects into smaller partitions to ensure we don't exceed the limitation
+    Lists.partition(new ArrayList<>(sObjects), 100).stream()
```

Comment:
extract 100 as constant and name appropriately. This makes it more clear what 100 is

Reply:
Done.